crates/sui-framework/packages/sui-system/sources/sui_system.move

```diff
@@ -555,6 +559,16 @@ module sui_system::sui_system {
         ctx: &mut TxContext,
     ) : Balance<SUI> {
         let self = load_system_state_mut(wrapper);
+
+        let previous_protocol_version = protocol_version(self);
+
+        // If we are moving from a version in which the AuthenticatorState object did not exist,
+        // to one where it does, create it!
+        if (previous_protocol_version < AuthenticatorStateFirstProtocolVersion
+            && next_protocol_version >= AuthenticatorStateFirstProtocolVersion) {
+            authenticator_state::create(ctx);
```

Comment:
We are shipping this with a different schedule to mainnet and testnet. But this will be always created at a particular protocol version?
An alternative way of doing this, which may be cleaner, is that in the Rust code when generating epoch change transaction, we could check whether the this feature is being enabled and if so, we insert a PT command into the epoch change transaction that creates the authenticator state.

Reply:
I looked at doing that and it was very awkward to get the previous protocol config, but I will take another look. I agree that we should probably do that since as you point out, doing this without flags is a bit problematic